### PR TITLE
Support for ray and offset-distance

### PR DESCRIPTION
**Pull request by DivyanshiVashist**

### CODE REVIEW

File: src/toTransform.js [hunk 1]
```diff
@@ -39,36 +39,77 @@
...
 
+  function convertRayString (properties) {
+    var offsetPath = internalScope.offsetPathParse(properties['offset-path']);
+
+    var offsetDistance = internalScope.offsetDistanceParse(properties['offset-distance']);
+    if (offsetDistance === undefined) {
+      offsetDistance = {value: 0, unit: 'px'};
+    }
+
+    pathElement.setAttribute('d', offsetPath.input);
```
Comment (ewilligers) [inline src/toTransform.js]: We shouldn't be setting d. Perhaps getOffsetDistanceLength should take a path length as input. We can pass in 0 for ray, with a FIXME to calculate path length based on <size>.

For now, all % lengths will become 0 for ray.

File: test/offsetRayDistanceTest.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+      assertTransformInterpolation([
+                                    {'offset-path': 'ray(-45deg)', 'offset-distance': '0px'},
+                                    {'offset-path': 'ray(-45deg)', 'offset-distance': '100px'}],
+        [
+                                    {at: 0, is: 'translate3d(0px, 0px, 0px)'},
+                                    {at: 1, is: 'translate3d(-70.71px, -70.71px, 0px)'}
+        ]
+      );
+    });
```
Comment (ewilligers) [inline test/offsetRayDistanceTest.js]: Add a test with 100% to verify we get 0x.

File: src/toTransform.js [hunk 3]
```diff
@@ -39,36 +39,77 @@
...
       return null;
     }
 
+    if (offsetPath.type === 'path') {
+      return convertPathString(properties);
+    }
+
+    if (offsetPath.type === 'ray') {
+      return convertRayString(properties);
+    }
```
Comment (alancutter) [inline src/toTransform.js]: This should return null for other types

File: src/toTransform.js [hunk 4]
```diff
@@ -39,36 +39,77 @@
...
+  }
+
+  function getOffsetDistanceLength (offsetDistance, pathElement) {
+    if (offsetDistance.unit === '%') {
+      return Number(offsetDistance.value) * pathElement.getTotalLength() / 100;
+    } else {
+      return Number(offsetDistance.value);
```
Comment (alancutter) [inline src/toTransform.js]: Optional: We shouldn't have to be converting to Number here, that should happen in the parse method.

File: src/toTransform.js [hunk 5]
```diff
@@ -39,36 +39,77 @@
...
+    }
+
+    pathElement.setAttribute('d', offsetPath.input);
+
+    var offsetDistanceLength = getOffsetDistanceLength(offsetDistance, pathElement);
+
+    var deltaX = Math.sin(offsetPath.input * Math.PI / 180) * offsetDistanceLength;
+    var deltaY = (-1) * Math.cos(offsetPath.input * Math.PI / 180) * offsetDistanceLength;
+
+    return 'translate3d(' + Math.round(deltaX * 100) / 100 + 'px, ' + Math.round(deltaY * 100) / 100 + 'px, 0px)';
```
Comment (alancutter) [inline src/toTransform.js]: Nit: Wrap the angle unit conversion and rounding to 100ths with helper functions to replace math with meaning.

File: src/toTransform.js [hunk 6]
```diff
@@ -39,36 +39,77 @@
...
+  }
+
   function convertOffsetAnchorPosition (properties, element) {
-    /* According to spec: https://drafts.fxtf.org/motion-1/#offset-anchor-property
-       If offset-anchor is set to auto then it will compute to the value of offset-position. */
+    // console.log("properties: " , properties + " element: " + element);
```
Comment (alancutter) [inline src/toTransform.js]: I don't think this should be here.

File: src/toTransform.js [hunk 7]
```diff
@@ -39,36 +39,77 @@
...
+    // According to spec: https://drafts.fxtf.org/motion-1/#offset-anchor-property
+    // If offset-anchor is set to auto then it will compute to the value of offset-position.
     var position = 'auto';
+
+    if (element === undefined) {
+      return null;
+    }
```
Comment (alancutter) [inline src/toTransform.js]: Is this related to your change?
